mlir/lib/Dialect/ArmSVE/Transforms/LegalizeForLLVMExport.cpp

```diff
@@ -66,6 +68,77 @@ using ScalableMaskedDivFOpLowering =
     OneToOneConvertToLLVMPattern<ScalableMaskedDivFOp,
                                  ScalableMaskedDivFIntrOp>;
 
+namespace {
+
+/// Unrolls a conversion to/from equivalent vector types, to allow using a
+/// conversion intrinsic that only supports 1-D vector types.
```

Comment:
That's the whole reason this op exists though? If this mapped 1-1 to the intrinsics there'd be no reason for these ops to exist.

Reply:
Yeah I understand that, but it doesn't mean both steps have to be done when lowering to intrinsics. At the Vector dialect level for example operations on higher rank vectors are typically broken down to rank 1 vectors first (VectorToSCF), before lowering to intrinsics.

Reply:
That'll be quite a big rework as the ArmSVE dialect is pretty much just a skeleton, so currently has no `-arm-sve-to-scf` (or similar) passes right now.

Reply:
I don't think `-arm-sve-to-scf` would make sense since this isn't emitting SCF ops, but I get your point, it would need some to live somewhere that doesn't exist today. Like I said, I'm happy with this regardless.

Reply:
I second Cullen here. It's not obvious to me what the right "pass" would be and where would it live, but we should add a TODO in the comments and in the commit message. Something along the lines:

> Extract the lowering of `convert_to_svbool` into a series of `vector.extract/insert` operations into a dedicated lowering layer/pass pre "LegalizeForLLVMExport". The latter should be as simple as 1-1 mapping.